src/ngraph/runtime/gpu/cuda_emitter.cpp

```diff
@@ -676,6 +680,236 @@ size_t runtime::gpu::CUDAEmitter::build_elementwise_n_to_1(const GPURuntimeConte
     return primitive_index;
 }
 
+size_t runtime::gpu::CUDAEmitter::build_replace_slice(const GPURuntimeContext* ctx,
+                                                      const std::array<std::string, 3>& dtypes,
+                                                      const Shape& tensor_shape,
+                                                      const Shape& source_shape,
+                                                      const Coordinate& lower_bounds,
+                                                      const Coordinate& upper_bounds,
+                                                      const Strides& slice_strides)
+{
+    // assumes NC{d1,...,dn} format
+    std::string kernel_name = "repslice";
+    std::stringstream ss;
+    ss << kernel_name << "_s" << join(tensor_shape, "_") << "_ssrc" << join(source_shape, "_")
+       << "_sll" << join(lower_bounds, "_") << "_slu" << join(upper_bounds, "_") << "_slst"
+       << join(slice_strides, "_");
+    auto hash = ss.str();
+
+    // check if the requested kernel is already an inserted primitive
+    size_t primitive_index = m_primitive_emitter->lookup(hash);
+    if (primitive_index != std::numeric_limits<size_t>::max())
+    {
+        return primitive_index;
+    }
+
+    constexpr const int nthreads_per_block = 32;
+
+    // if the kernel has not been compiled, build it
+    auto compiled_kernel = ctx->compiled_kernel_pool->get(kernel_name);
```

Comment:
if kernel use different data type, the kernel_name still the same, it will get the same compiled kernel which sounds not right.

Reply:
You are correct. I'm still not sure if this is really ever a concern, switching types in the middle of a function, but best to be safe.

Reply:
I have added the dtypes to the kernel_name.

Reply:
COOL. The kernel should be able to shared between functions.